### PR TITLE
GM16/Republicize

### PR DESCRIPTION
**Pull request by artpi**
Introduces functionality to share messages on social media multiple times
Part of GM16 project

##Testing 

- Connect twitter / facebook, etc
- Go to posts list
- Edit published post
- go to sharing
- See "Share" button
- Click it
- See that message goes to your connected social media
- Return to posts lict
- See megaphone at the bottom of any published post
- Click it
- See similar functionality
- Edit mssage
- Click "Share again"
- See that it goes out to your twitter / etc
- Dismiss success message

## Visuals

zrzut ekranu 2016-09-17 o 16 50 02

### CODE REVIEW

**Comment by mtias** [comment]
Break it down a bit? :)

File: client/my-sites/posts/post-share.jsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+	getInitialState() {
+		return {
+			skipped: PostMetadata.publicizeSkipped( this.props.post ) || [],
+			message: PostMetadata.publicizeMessage( this.props.post )
+		}
+	},
+
+	hasConnections: function() {
+		return this.props.connections && this.props.connections.length;
```
Comment (enejb) [inline client/my-sites/posts/post-share.jsx]: Does this function return a bool or an int?

File: client/my-sites/posts/post-share.jsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+		const classes = classNames( 'posts__post-share', {
+			'has-connections': this.hasConnections()
+		} );
+
+		return (
+			<div className="posts__post-share-wrapper">
+				{ this.props.requesting && <Notice status="is-warning" showDismiss={ false }>{ this.translate( 'Hang tight, socializing your media...' ) }</Notice> }
+				{ this.props.success && <Notice status="is-success" onDismissClick={ this.dismiss }>{ this.translate( 'It went out! Your social media is on fire!' ) }</Notice> }
+				{ this.props.failure && <Notice status="is-error" onDismissClick={ this.dismiss }>{ this.translate( 'Something went wrong. Please dont be mad.' ) }</Notice> }
```
Comment (enejb) [inline client/my-sites/posts/post-share.jsx]: Typo dont -> don't

File: client/my-sites/posts/style.scss [hunk 3]
```diff
@@ -23,6 +23,9 @@
 		@include breakpoint( "<660px" ) {
 			font-size: 13px;
 		}
+		.posts__post-share-wrapper .notice {
+			margin-bottom: 0
```
Comment (enejb) [inline client/my-sites/posts/style.scss]: I think this could use `;` at the end.

File: client/post-editor/editor-sharing/publicize-options.jsx [hunk 4]
```diff
@@ -180,6 +183,24 @@ const EditorSharingPublicizeOptions = React.createClass( {
 		);
 	},
 
+	renderRepublicize() {
+		return(
+			<Button
+				className="button editor-sharing__publicize-share-button"
+				disabled={ ( this.props.connections.length - PostMetadata.publicizeSkipped( this.props.post ).length < 1 ) || this.props.requesting }
+				onClick={ () => {
```
Comment (enejb) [inline client/post-editor/editor-sharing/publicize-options.jsx]: Would it make more sense to break this up into its own function?

File: client/state/sharing/publicize/reducer.js [hunk 5]
```diff
@@ -13,10 +13,27 @@ import {
...
 import { connectionsSchema } from './schema';
 import { isValidStateWithSchema } from 'state/utils';
+import { createReducer } from 'state/utils';
+
+
+const getStateWithSharePostFething = ( state, fetching, siteId, postId ) => Object.assign( {}, state, {
+	[ siteId ]: Object.assign( {}, state[ siteId ], { [ postId ]: fetching } )
+} );
+const sharePostStatus = createReducer( {}, {
+	[ PUBLICIZE_SHARE ]: ( state, action ) => getStateWithSharePostFething( state, { requesting: true }, action.siteId, action.postId ),
```
Comment (enejb) [inline client/state/sharing/publicize/reducer.js]: typo: getStateWithSharePostFething => getStateWithSharePostFetching

File: config/wpcalypso.json [hunk 6]
```diff
@@ -100,6 +100,7 @@
 		"reader/tags-with-elasticsearch": true,
 		"reader/refresh-2016-07": true,
 		"resume-editing": true,
+		"republicize": false,
```
Comment (enejb) [inline config/wpcalypso.json]: Lets set this to true.

File: config/development.json [hunk 7]
```diff
@@ -124,6 +124,7 @@
 		"reader/start": true,
 		"reader/tags-with-elasticsearch": true,
 		"resume-editing": true,
+		"republicize": false,
```
Comment (enejb) [inline config/development.json]: Set to true. so that we can remove the flag

File: client/my-sites/posts/post-share.jsx (hunk 1 above)
Comment (artpi) [inline client/my-sites/posts/post-share.jsx]: Fixed

File: client/my-sites/posts/post-share.jsx (hunk 2 above)
Comment (artpi) [inline client/my-sites/posts/post-share.jsx]: Fixed

File: client/my-sites/posts/style.scss (hunk 3 above)
Comment (artpi) [inline client/my-sites/posts/style.scss]: Fixed

File: client/post-editor/editor-sharing/publicize-options.jsx (hunk 4 above)
Comment (artpi) [inline client/post-editor/editor-sharing/publicize-options.jsx]: Fixed

File: client/state/sharing/publicize/reducer.js (hunk 5 above)
Comment (artpi) [inline client/state/sharing/publicize/reducer.js]: Fixed

File: config/development.json (hunk 7 above)
Comment (artpi) [inline config/development.json]: Enabled

File: config/wpcalypso.json (hunk 6 above)
Comment (artpi) [inline config/wpcalypso.json]: Enabled

**Comment by artpi** [comment]
P2 and relevant links: p58i-41F-p2

**Pull request merged**